Sources/XMTPiOS/Libxmtp/DecodedMessageV2.swift

```diff
@@ -0,0 +1,506 @@
+import Foundation
+import LibXMTP
+
+public struct DecodedMessageV2: Identifiable {
+    private let ffiMessage: FfiDecodedMessage
+
+    public var id: String {
+        ffiMessage.id().toHex
+    }
+
+    // Note: conversationId is not directly available in FfiDecodedMessage
+    // It would need to be passed in or derived from context
+
+    public var senderInboxId: InboxId {
+        ffiMessage.senderInboxId()
+    }
+
+    public var sentAt: Date {
+        Date(timeIntervalSince1970: TimeInterval(ffiMessage.sentAtNs()) / 1_000_000_000)
+    }
+
+    public var sentAtNs: Int64 {
+        ffiMessage.sentAtNs()
+    }
+
+    public var deliveryStatus: MessageDeliveryStatus {
+        switch ffiMessage.deliveryStatus() {
+        case .unpublished:
+            return .unpublished
+        case .published:
+            return .published
+        case .failed:
+            return .failed
+        }
+    }
+
+    // Topic generation requires conversationId which is not available in FfiDecodedMessage
+    public var topic: String {
+        "" // Would need to be passed in or derived from context
+    }
+
+    public var reactions: [DecodedMessageV2]? {
+        let reactionMessages = ffiMessage.reactions()
+        guard !reactionMessages.isEmpty else { return nil }
+        return reactionMessages.compactMap { DecodedMessageV2(ffiMessage: $0) }
+    }
+
+    public func content<T>() throws -> T {
```

Comment:
One thing that's controversial here is that we don't have an `encodedContent` field. I think that's a cleaner separation, and is certainly more memory efficient if we can throw out the original payload. Not sure who was using it. We pull out the content type, fallback, etc and have that available separately.